src/v/cluster/controller_backend.cc

```diff
@@ -245,27 +245,22 @@ std::error_code check_configuration_update(
 } // namespace
 
 struct controller_backend::ntp_reconciliation_state {
-    std::optional<model::revision_id> changed_at;
+    // A counter to avoid losing notifications that arrived concurrently with
+    // reconciliation.
+    size_t notify_count = 0;
```

Comment:
nit: maybe a name `pending_notifies` would better reflect the purpose of this variable. It looks like the state is reconciled if there are no pending notifies.

Reply:
right, but then we'll need to subtract the counter in `mark_reconciled` to really keep it "pending"